frontend/src/components/Users/Users.tsx

```diff
@@ -1,45 +1,70 @@
-import React from 'react'
-import { List, ListItem, ListItemIcon, ListItemText, Divider } from '@material-ui/core'
+import React, { useEffect } from 'react'
+import { List, ListItemIcon, ListItemText, Divider,  } from '@material-ui/core'
 import { Duration } from '../Duration'
 import { Platform } from '../Platform'
+import { ApplicationState } from '../../store'
+import { useSelector } from 'react-redux'
+import { ListItemLocation } from '../ListItemLocation/ListItemLocation'
+import { ShareDetails } from '../DeviceShareContainer/ContactCardActions'
+import { SharingManager } from '../../services/SharingManager'
+import { getUsersConnectedDevice } from '../../models/devices'
 
 interface Props {
+  deviceId: string
   service?: IService | null
-  connected?: boolean
 }
 
-export const Users: React.FC<Props> = ({ service, connected }) => {
-  if (!service?.access.length) return null
+export const Users: React.FC<Props> = ({ deviceId, service }) => {
+  const [shares, setShares] = React.useState<ShareInfo[]>([])
+  const devices = useSelector((state: ApplicationState) => state.devices.all)
+  const findDevice = (id: string) => devices.find((d: IDevice) => d.id === id)
+  const device = findDevice(deviceId)
+  const shared = service ? service?.access.length : device?.access.length
+  const usersConnected = getUsersConnectedDevice(device)
 
-  const users = connected
-    ? service.sessions
-    : service.access.filter(user => !service.sessions.some(session => user.email === session.email))
```

Comment:
Looks like we lost some important functionality here. Let's discuss. See below:
<img width="1509" alt="Screen Shot 2020-07-30 at 4 20 30 AM" src="https://user-images.githubusercontent.com/864957/88917180-334fb380-d21c-11ea-860d-3b13e77e4d62.png">